rust/src/connection.rs

```diff
@@ -1,33 +1,79 @@
-use crate::wallet;
+use crate::config;
+use crate::lightning::PeerManager;
 use crate::wallet::is_first_channel_usable;
-use anyhow::Result;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::task::JoinHandle;
 
-/// blocks the current thread, if you want this to be run asynchronously you need to spawn a thread
-/// from the outside.
-pub async fn keep_alive() -> Result<()> {
-    let mut connected = false;
-    loop {
-        if !connected || !is_first_channel_usable() {
-            connected = connect().await?;
-        }
-
-        // looping here indefinitely to keep the connection with the maker alive.
-        tokio::time::sleep(std::time::Duration::from_secs(5)).await;
-    }
+pub struct Connection {
```

Comment:
yes it is not necessary, but it reflects a bit my preferences in code style coming from object oriented languages. I like methods better than functions. However its not needed here. What is the rust idiomatic way of deciding when to put something into a method or function?

Reply:
> What is the rust idiomatic way of deciding when to put something into a method or function?

I'm not sure there is such a thing. I think it's a matter of style. I like to use methods when it reads better e.g.

```rust
actor.do(action) > do(actor, action)
```

In this case I would take issue with _introducing_ an internal type just so that we can call a method on it. I'm not sure the `Connection` needs to be modeled in this case. 

It's also interesting that the type only holds the `PeerManager`, which I would argue is not "the connection". If it were holding the `peer_pubkey`, `peer_addr` and the `RemoteHandle` to the spawned task (to be able to cancel the task on drop for example) then it might make more sense.

Reply:
fair enough - see 393e8d4fd730f5b397e54256a051f7e88406b49f